hydra-node/src/Hydra/Chain/Direct/Tx.hs

```diff
@@ -552,55 +556,48 @@ closeTx scriptRegistry vk closing startSlotNo (endSlotNo, utcTime) openThreadOut
   headScript =
     fromPlutusScript @PlutusScriptV2 Head.validatorScript
 
-  closeVersion
-    | offChainVersion == version && toInteger version == snapshotNumber && snapshotNumber == 0 = Head.InitialVersion
-    | offChainVersion == version = Head.CurrentVersion
-    | offChainVersion == version + 1 = Head.OutdatedVersion
-    | otherwise = Head.InitialVersion
-
-  headRedeemer =
-    toScriptData
-      Head.Close
-        { signature
-        , version = closeVersion
-        , utxoToDecommitHash =
-            case closeVersion of
-              Head.CurrentVersion ->
-                toBuiltin $ hashUTxO @Tx mempty
-              Head.OutdatedVersion ->
-                toBuiltin decommitUTxOHashBytes
-              Head.InitialVersion ->
-                toBuiltin $ hashUTxO @Tx mempty
-        }
+  headRedeemer = toScriptData $ Head.Close closeRedeemer
+
+  -- TODO: push this further out to avoid 'offChainVersion'?
+  closeRedeemer
+    | offChainVersion == version && toInteger version == snapshotNumber && snapshotNumber == 0 =
+        Head.CloseInitial
+    | offChainVersion == version =
+        Head.CloseCurrent{signature}
+    | offChainVersion == version + 1 =
+        Head.CloseOutdated
+          { signature
+          , alreadyDecommittedUTxOHash = toBuiltin decommitUTxOHashBytes
+          }
+    | otherwise = Head.CloseInitial
```

Comment:
what if we use a `Head.CloseOutdated` instead?

Reply:
Same result, both won't work.